### PR TITLE
Fixes #25820: On new groups, adding a prop leads to "No query defined for group" error

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/25820

Correct the case where updating a dynamic group with empty query was in error and also make a bit clearer the case where there is no criterion in the UI: 

When group is created: 
image

When we click on the plus: 
image

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/components/SearchNodeComponent.scala [hunk 1]
```diff
@@ -321,7 +323,11 @@ class SearchNodeComponent(
         ("id"    -> "SubmitSearch"),
         ("class" -> "btn btn-primary ")
       ) &
-      "#query_lines *" #> criteria.zipWithIndex.flatMap { case (cl, i) => displayQueryLine(cl, i, criteria.size > 1) }).apply(
+      "#query_lines *" #> (if (criteria.isEmpty) <div><span>Add a first criterion line to define your group: </span>{
```
Comment (amousset) [inline webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/components/SearchNodeComponent.scala:326]:
```suggestion
      "#query_lines *" #> (if (criteria.isEmpty) <div><span>Add a first criterion to define your group: </span>{
```

**Review by clarktsiory — COMMENTED** [review]
LGTM and nominal case works, but I think the "Search" button needs to be hidden, or else making a search without any criteria and saving will make the group have "all nodes"

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
The search button is now hidden when group is created.

**Comment by fanf** [comment]
PR rebased

**Pull request merged**